students/WooseokJ/MailP2fixedWJ.py

```diff
@@ -0,0 +1,111 @@
+
+# coding: utf-8
+
+# In[3]:
+
+
+# print of the main menu
+import sys
+import datetime
+import operator
+
+def menu():
+    print(''''Menu
+    Please choose from below options:
+    1 - Send a Thank You
+    2 - Create a Report
+    3 - Exit the program
+    4 - Print thank you letter to all current doners''')
+#first it will ask to choose a name whoom you would want to send the mail
+#along side that when typing list it will give you all the names on the current list
+#second if the name is not in the list then it will prompt to add the name into the data base
+#lastly it will print the message of the email
+# have a problem here where when asked for the amount of donation if the value type is not a number we get a error how to loop around this?
+def send_a_thankyou():
+    donor_list = []
+    for donor in donor_db:
+        donor_list.append(donor.lower())
+    while True:
+        fullname = input("Enter the full name of the donor typing list will give the names of current doners ")
+        if fullname.lower() == 'list':
+            for donor in donor_db:
+                print(donor)
+        elif fullname.lower() not in donor_list:
+            print(f'{fullname} does not exist in the donor data, '
+                  f'we will add {fullname} to the donor data.')
+            amount = float(input("Please enter the donation amount: "))
+            donor_db.append((fullname, [amount, 1]))
```

Comment:
For this one, remember _donor_db_ is a dictionary, not a list, so it doesn't have an _append()_ method. You should be able to do _donor_db[fullname].append(...)_ since that would be calling the list of donations associated with a particular donor.